packages/core-plugin-api/src/adaptable-components/types.ts

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ComponentType, RefAttributes } from 'react';
+
+import { BackstagePlugin } from '../plugin';
+
+/** @public */
+export type OpaqueComponentProps<Context extends {}> = { value: Context };
```

Comment:
Please prefix type parameters with `T*`, https://github.com/backstage/backstage/blob/master/STYLE.md#naming

Reply:
Yep everything, the `T*` is there to make it easier to read the type definitions without wondering what's an arg and what isn't.